internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/spaces.go

```diff
@@ -125,6 +125,31 @@ func (h *Handler) addSpaceMember(w http.ResponseWriter, r *http.Request, info *p
 			Nanos:   uint32(expiration.UnixNano() % int64(time.Second)),
 		}
 	}
+
+	ref := provider.Reference{ResourceId: info.GetId()}
+	p, err := h.findProvider(ctx, &ref)
+	if err != nil {
+		response.WriteOCSError(w, r, response.MetaNotFound.StatusCode, "error getting storage provider", err)
+		return
+	}
+
+	providerClient, err := h.getStorageProviderClient(p)
+	if err != nil {
+		response.WriteOCSError(w, r, response.MetaNotFound.StatusCode, "error getting storage provider client", err)
+		return
+	}
+
+	lgRes, err := providerClient.ListGrants(ctx, &provider.ListGrantsRequest{Ref: &ref})
+	if err != nil || lgRes.Status.Code != rpc.Code_CODE_OK {
+		response.WriteOCSError(w, r, response.MetaServerError.StatusCode, "error listing space grants", err)
+		return
+	}
+
+	if len(lgRes.Grants) == 1 || !isSpaceManagerRemaining(lgRes.Grants, grantee) {
```

Comment:
This means I cannot add a member to my space when there is only one member. 
```suggestion
	if !isSpaceManagerRemaining(lgRes.Grants, grantee) {
```
`isSpaceManagerRemaining` assumes that the grantee does not have manager permissions. This should be fine in most cases but we will do unnecessary requests. Maybe we skip all checks when the grantee gets Manager permissions?